### PR TITLE
Enforce nexus request timeout in workflow test suite

### PR DESCRIPTION
**Pull request by bergundy**
Also fixes issue where headers injected by workflow outbound interceptors in the test env were not normalized to lower case as done by the real server.

### CODE REVIEW

File: internal/internal_workflow_testsuite.go [hunk 1]
```diff
@@ -2354,6 +2354,17 @@ func (env *testWorkflowEnvironmentImpl) newTestNexusTaskHandler() *nexusTaskHand
...
+	// Use lower case header values to simulate how the Nexus SDK (used internally by the "real" server) would transmit
+	// these headers over the wire.
+	nexusHeader := make(map[string]string, len(params.nexusHeader))
+	for k, v := range params.nexusHeader {
+		nexusHeader[strings.ToLower(k)] = v
+	}
+	params.nexusHeader = nexusHeader
+	// The real server allows requests to take up to 10 seconds, mimic that behavior here.
+	// Note that if a user sets the Request-Timeout header, it gets overridden.
+	params.nexusHeader[strings.ToLower(nexus.HeaderRequestTimeout)] = "10s"
```
Comment (rodrigozhou) [inline internal/internal_workflow_testsuite.go:2366]: We also set the request timeout for the cancel operation.

Comment (bergundy) [reply internal/internal_workflow_testsuite.go:2366]: Yes, this will set that timeout for all "requests".

**Comment by Quinn-With-Two-Ns** [comment]
Seems like there is an issue with this PR only on Windows

**Comment by bergundy** [comment]
> Seems like there is an issue with this PR only on Windows

Changed the test slightly, CI should pass now AFAICT.

**Comment by bergundy** [comment]
Looks like this is good to go @Quinn-With-Two-Ns.

File: internal/internal_workflow_testsuite.go [hunk 2]
```diff
@@ -2358,6 +2358,17 @@ func (env *testWorkflowEnvironmentImpl) ExecuteNexusOperation(
...
```
Comment (Quinn-With-Two-Ns) [inline internal/internal_workflow_testsuite.go:2370]: Isn't the real server behaviour the minimum of 10s or the remaining schedule to close?

Comment (bergundy) [reply internal/internal_workflow_testsuite.go:2370]: Good point. Not at the moment, but we may enforce that eventually.

**Review by Quinn-With-Two-Ns — APPROVED** [review]

**Pull request merged**